packages/server/src/FlowgladServer.ts

```
@@ -350,11 +352,64 @@ export class FlowgladServer {
       usageEvent: {
         ...parsedParams,
         properties: parsedParams.properties ?? undefined,
-        usageDate: parsedParams.usageDate || undefined,
+        usageDate: parsedParams.usageDate ?? undefined,
       },
     })
   }
 
+  /**
+   * Create multiple usage events in a single request.
+   * NOTE: this method is more efficient than calling `createUsageEvent` multiple times.
+   * @param params - The parameters containing an array of usage events.
+   * @returns The created usage events.
+   * @throws {Error} If any subscription in the bulk request is not owned by the authenticated customer.
+   */
+  public createBulkUsageEvents = async (
+    params: CreateBulkUsageEventsParams
+  ): Promise<{
+    usageEvents: FlowgladNode.UsageEvents.UsageEventCreateResponse['usageEvent'][]
+  }> => {
+    const parsedParams = createBulkUsageEventsSchema.parse(params)
+
+    // Get the scoped customer to validate ownership
+    const { customer } = await this.getCustomer()
+
+    // Extract unique subscription IDs to validate ownership
+    const uniqueSubscriptionIds = [
+      ...new Set(
+        parsedParams.usageEvents.map((e) => e.subscriptionId)
+      ),
+    ]
+
+    // Batch fetch all subscriptions to validate ownership
+    const subscriptions = await Promise.all(
+      uniqueSubscriptionIds.map((id) =>
+        this.flowgladNode.subscriptions.retrieve(id)
+      )
+    )
```

Let's check this against billing.currentSubscriptions.map(sub => sub.id) so that we don't need to make an API hit.

And then we can throw an error if we received an event with a sub id not found among the customer's current subs